test/msw-api/setup-worker/start/on-unhandled-request/suggestions.test.ts

```diff
@@ -0,0 +1,290 @@
+import * as path from 'path'
+import {
+  captureConsole,
+  filterLibraryLogs,
+} from '../../../../support/captureConsole'
+import { runBrowserWith } from '../../../../support/runBrowserWith'
+
+function createRuntime() {
+  return runBrowserWith(path.resolve(__dirname, 'suggestions.mocks.ts'))
+}
+
+describe('REST API', () => {
+  test('does not suggest any handlers when there are no similar ones', async () => {
+    const runtime = await createRuntime()
+    const { messages } = captureConsole(runtime.page)
+
+    await runtime.request({
+      url: runtime.makeUrl('/user-details'),
+    })
+
+    const libraryWarnings = messages.warning.filter(filterLibraryLogs)
+    const unhandledRequestWarning = libraryWarnings.find((text) => {
+      return /\[MSW\] Warning: captured a request without a matching request handler/.test(
+        text,
+      )
+    })
+
+    expect(unhandledRequestWarning).toMatch(`\
+[MSW] Warning: captured a request without a matching request handler:
+
+  • GET /user-details
+
+If you still wish to intercept this unhandled request, please create a request handler for it. Read more: https://mswjs.io/docs/getting-started/mocks.`)
+
+    return runtime.cleanup()
+  })
+
+  test('suggests a similar request handler of the same method', async () => {
+    const runtime = await createRuntime()
+    const { messages } = captureConsole(runtime.page)
+
+    await runtime.request({
+      url: runtime.makeUrl('/users'),
+    })
+
+    const libraryWarnings = messages.warning.filter(filterLibraryLogs)
+    const unhandledRequestWarning = libraryWarnings.find((text) => {
+      return /\[MSW\] Warning: captured a request without a matching request handler/.test(
+        text,
+      )
+    })
+
+    expect(unhandledRequestWarning).toMatch(`\
+[MSW] Warning: captured a request without a matching request handler:
+
+  • GET /users
+
+Did you mean to request "GET /user" instead?
```

Comment:
```suggestion
Did you mean to intercept the "GET /user" request instead?
```

Reply:
That's not equivalent. The suggestion hints that you've requested a different resource. It treats an actual request as potentially invalid, not the handler. 

Given you have a `rest.get('/user')` handler but performed a `GET /users` request, the library will suggest that maybe you meant to request `GET /user`, because you have that handled.

Reply:
Thanks for taking the time to respond 😄